packages/general-settings/middleware.js

```diff
@@ -1,12 +1,34 @@
 import { getURL } from '@bufferapp/publish-formatters';
+import { actionTypes as profileActionTypes } from '@bufferapp/publish-profile-sidebar';
+import {
+  actions as asyncDataFetch,
+  actions as dataFetchActions,
+} from '@bufferapp/async-data-fetch';
 import { actionTypes } from './reducer';
 
-export default () => next => (action) => {
+export default ({ dispatch }) => next => (action) => {
   next(action);
   switch (action.type) {
     case actionTypes.SET_DIRECT_POSTING:
       window.location = getURL.getInstagramDirectPostingURL(action.profileId);
       break;
+    case profileActionTypes.SELECT_PROFILE:
+      dispatch(dataFetchActions.fetch({
+        name: 'getLinkShortener',
+        args: {
+          profileId: action.profile.id,
+        },
```

Comment:
Can you get the `profileId` directly out of action here, like:
```suggestion
        profileId: action.profileId
```